src/Endpoint/VMware.ScriptRuntimeService.RunspaceClient/RunspaceClient.cs

```diff
@@ -17,7 +17,32 @@ namespace VMware.ScriptRuntimeService.RunspaceClient {
    /// Runspace Endpoint API Client
    /// </summary>
    public class RunspaceClient : IRunspace {
-      private ScriptApi _scriptsApiClient;
+      private readonly ScriptApi _scriptsApiClient;
+
+      static RunspaceClient() {
+         // Override Newtonsoft.Json MaxDepth de/serialization because
+         // as of version 13.0.1 the default MaxDepth is 64
+         // this keeps pre 13.0.1 behaviour
+         Func<Newtonsoft.Json.JsonSerializerSettings> defaultSettingsFunc =
+            Newtonsoft.Json.JsonConvert.DefaultSettings;
+         if (null == Newtonsoft.Json.JsonConvert.DefaultSettings) {
+            defaultSettingsFunc =
+               () => new Newtonsoft.Json.JsonSerializerSettings() {
+                  MaxDepth = int.MaxValue,
+                  NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore
+               };
+         } else {
+            defaultSettingsFunc =
+               () => {
+                  var settings = defaultSettingsFunc();
+                  settings.MaxDepth = int.MaxValue;
+
+                  return settings;
+               };
+         }
+
+         Newtonsoft.Json.JsonConvert.DefaultSettings = defaultSettingsFunc;
```

Comment:
If code in other classes depend on this global setting, they might not be affected if they don't trigger this static ctor.
Consider placing this in some initialization logic that is explicit or even better - don't use the global settings but pass them where you use the serializers.

Reply:
The whole library revolves around RunspaceClient and has no other use-cases.